### DISCUSSION TITLE
[VTM] offline map loading failure, tileSource.getMapInfo() returning null

### DISCUSSION THREAD

**Discussion by deepak07dbz** [mapsforge/vtm#1118] [Q&A] [answered]
void loadMap(File areaFolder) {
//map event receiver
mapView.map().layers().add(new MapEventsReceiver(mapView.map()));

    //map file source
    MapFileTileSource tileSource = new MapFileTileSource();
    File mapFile = new File(areaFolder, AREA + ".map").getAbsoluteFile();
    tileSource.setMapFile(String.valueOf(mapFile));
    Log.d("LOAD", "loadMap: " + mapFile.exists());
    VectorTileLayer l = mapView.map().setBaseMap(tileSource);
    mapView.map().setTheme(VtmThemes.DEFAULT);
    mapView.map().layers().add(new BuildingLayer(mapView.map(), l));
    mapView.map().layers().add(new LabelLayer(mapView.map(), l));

    //marker's layer
    itemLayer = new ItemizedLayer<>(mapView.map(), (MarkerSymbol) null);
    mapView.map().layers().add(itemLayer);

    //map position
    MapInfo mapInfo = tileSource.getMapInfo();
    if (mapInfo == null) {
        Log.d("LOAD", "loadMap: null mapinfo");
    } else if (mapInfo.boundingBox == null) {
        Log.d("LOAD", "loadMap: bbox is null");
    } else {
        GeoPoint mapCenter = tileSource.getMapInfo().boundingBox.getCenterPoint();
        mapView.map().setMapPosition(mapCenter.getLatitude(),
mapCenter.getLongitude(), 1 << 17);
        setContentView(mapView);
    }
}

//mapFile exists and is a valid file

**Comment by devemux86** [comment, marked as answer]
See the [MapsforgeActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MapsforgeActivity.java) where the map info is used and works.

**Reply by deepak07dbz** [reply]
i want to use vtm's (version 0.13.0 ) tileSource.setMapFile method instead of setMapFileInputStream, is it possible

**Reply by devemux86** [reply]
It works the same. The map file must be valid and contain map info.

> version 0.13.0

This is very old version, please use latest version.

**Reply by devemux86** [reply]
> setMapFile

This is used in [MapsforgeTest](https://github.com/mapsforge/vtm/blob/master/vtm-playground/src/org/oscim/test/MapsforgeTest.java) and map info works for tileSource.getBoundingBox().

**Comment by deepak07dbz** [comment]
i used the latest version(0.21.0) also still the same error comes, also this is and android application.

**Reply by devemux86** [reply]
VTM samples work correctly on Android and desktop,
so could be an invalid map file or something in the app code.

Better check the VTM samples about how to use the library:
[GettingStarted](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/GettingStarted.java), [MapsforgeActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MapsforgeActivity.java), [MapsforgeTest](https://github.com/mapsforge/vtm/blob/master/vtm-playground/src/org/oscim/test/MapsforgeTest.java)

**Comment by deepak07dbz** [comment]
these examples work with uri and setMapFileInputStream, but i dont want to use these.

**Reply by deepak07dbz** [reply]
public void loadMap(File areaFolder, MapActivity activity)
  {
    logUser(activity, "loading map");

    // Map events receiver
    mapView.map().layers().add(new MapEventsReceiver(mapView.map()));

    // Map file source
    tileSource = new MapFileTileSource();
    tileSource.setMapFile(new File(areaFolder, currentArea + ".map").getAbsolutePath());
    VectorTileLayer l = mapView.map().setBaseMap(tileSource);
    mapView.map().setTheme(VtmThemes.DEFAULT);
    mapView.map().layers().add(new BuildingLayer(mapView.map(), l));
    mapView.map().layers().add(new LabelLayer(mapView.map(), l));
    
    // Markers layer
    itemizedLayer = new ItemizedLayer<>(mapView.map(), (MarkerSymbol) null);
    mapView.map().layers().add(itemizedLayer);
    customLayer = new ItemizedLayer<>(mapView.map(), (MarkerSymbol) null);
    mapView.map().layers().add(customLayer);

    // Map position
    GeoPoint mapCenter = tileSource.getMapInfo().boundingBox.getCenterPoint();
    mapView.map().setMapPosition(mapCenter.getLatitude(), mapCenter.getLongitude(), 1 << 12);

    ViewGroup.LayoutParams params =
        new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
    activity.addContentView(mapView, params);
    
    loadGraphStorage(activity);
  }
  this is from PocketMaps project, my project usage is similar to this.

**Reply by deepak07dbz** [reply]
https://github.com/junjunguo/PocketMaps/blob/master/PocketMaps/app/src/main/java/com/junjunguo/pocketmaps/map/MapHandler.java

**Reply by devemux86** [reply]
The library works the same with file or stream and the map info is correct (if map file is valid).

It is better to debug your code and study the VTM samples that work.

**Reply by deepak07dbz** [reply]
Thank you, I guess i found the issue, the app needs to download and store map file in app specific directory or ask user to select the map file.
